cmd/skopeo/main.go

```diff
@@ -99,13 +100,18 @@ func createApp() (*cli.App, *globalOptions) {
 }
 
 // before is run by the cli package for any command, before running the command-specific handler.
-func (opts *globalOptions) before(_ *cli.Context) error {
+func (opts *globalOptions) before(ctx *cli.Context) error {
 	if opts.debug {
 		logrus.SetLevel(logrus.DebugLevel)
 	}
 	if opts.tlsVerify.present {
 		logrus.Warn("'--tls-verify' is deprecated, please set this on the specific subcommand")
 	}
+	switch ctx.Args().First() {
+	case "", "help", "version":
+		return nil
+	}
+	unshare.MaybeReexecUsingUserNamespace(false)
```

Comment:
It’s great to have an one-liner 🎉 

Hard-coding the list of commands to ignore in here is acceptable but not ideal.  I wonder whether sticking this into maybe `opts.newSystemContext` would be a bit better — it would certainly not be _clean_ either (notably because that function can be called multiple times per process, and not necessarily at the immediate start of an operation), but it would be better targeted even with the current codebase (`manifest-digest`, `standalone-sign`, `standalone-verify` and `untrusted-signature-dump-without-verification` would be excluded).

Or, well, add a mandatory `shouldRunInUserNamespaces` return value from each of the `*Cmd` functions currently returning `cli.Command` and be explicit about it.

(BTW, `cli.App.Before` is not called for the built-in `help` and `version` subcommands anyway. But that’s undocumented AFAICS, so we should not rely on it.)

Reply:
I've whitelisted the commands that need a namespace using `cli.Command.Before`

Reply:
Nice idea. Can the code in `globalOptions.before` can be removed now?

Reply:
yes sure, forgot to amend it.  Done now